doc/arch/adr-003-collections-api.md

```diff
@@ -0,0 +1,82 @@
+| 4                |    7.15 |    3.83 |            2.83 |
+| 5                |    6.59 |    3.77 |            2.58 |
+
+| nlcd-soils-huc8 | develop | staging | collections-api |
+|-----------------|--------:|--------:|----------------:|
+| 1               |   31.18 |   16.48 |            8.42 |
+| 2               |   26.28 |   14.98 |            6.02 |
+| 3               |    31.1 |   14.36 |            4.89 |
+| 4               |    27.2 |   15.05 |            4.51 |
+| 5               |   29.27 |   12.48 |            4.09 |
+
+| average          | develop | staging | collections-api |
+|------------------|--------:|--------:|----------------:|
+| nlcd-soils-huc12 |   3.224 |   2.444 |           1.598 |
+| nlcd-soils-huc10 |   7.716 |   4.528 |           3.776 |
+| nlcd-soils-huc8  |  29.006 |   14.67 |           5.586 |
+
+Here's a chart of the table above:
+
+![image](https://cloud.githubusercontent.com/assets/1430060/26412141/c6eb7e26-4075-11e7-9b65-207bb286c171.png)
+
+The difference in `develop` and staging are because of two factors: staging workers are more powerful, and have much faster access to S3 tiles. These advantages should carry over to the Collections API variant as well, thus making it even faster when deployed.
+
+In addition, this will make our code simpler to understand and maintain, as well as make provisioning and deployment simpler by removing Spark and Spark JobServer.
+
+These improved speeds may also allow us to increase the area of interest threshold on the front-end, allowing users to select even bigger areas than before and get faster results back.
+
+## Challenges
+
+GeoTrellis 1.0 is compiled for Java 8, whereas the current MMW environment is setup with Java 7. We will have to make a number of updates to the MMW environment to support Java 8.
+
+Since Spark JobServer is no longer used, we will have to switch to using Akka HTTP for exposing the Geoprocessing service. This will also require updates to the relevant geoprocessing tasks in Celery.
+
+Changes to the geoprocessing service may result in differences in how cells are counted, and may lead to different results for the same inputs. This is already seen to some extent in [mmw-geoprocessing#44](https://github.com/WikiWatershed/mmw-geoprocessing/pull/44). It is difficult to say which is _more_ correct, the current or the new implementation. But we will have to explain and justify the changes.
+
+The Collections API has so far never been deployed to production. It comes with risks of "unknown unknowns", and we may have to solve unforeseen technical challenges.
+
+## Drawbacks
+
+Removing Spark JobServer will lead to the interaction between Celery and Geoprocessing becoming synchronous rather than asynchronous. This should be a tolerable difference, since the new synchronous requests should be 3–5 times faster than the old asynchronous ones, and a Celery task waiting for 5-10 seconds should be manageable.
```

Comment:
Agreed with converting these calls to be synchronous. In addition, I think that we should cap the HTTP request timeout to the backend geoprocessing service to some number of seconds and mark the request as a failure if it exhausts the timeout.